spec/javascripts/components/accordion-spec.js

```diff
@@ -49,22 +63,50 @@ describe('Accordion component', function () {
 
   afterEach(function () {
     document.body.removeChild(container)
+    if (GOVUK.analytics.trackEvent.calls) {
+      GOVUK.analytics.trackEvent.calls.reset()
+    }
+  })
+
+  it('applies custom attributes to click event if specified in section', function () {
+    startAccordion()
+    document.querySelector('.govuk-accordion__section-header button').click()
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('pageElementInteraction', 'accordionOpened', { transport: 'beacon', label: 'Accordion Section Heading Text', dimension114: 1 })
+  })
+
+  it('applies custom attributes to click event if specified in show/hide section', function () {
+    var object = {
+      'track-options': {
+        dimensionExample: 'Example value'
+      }
+    }
+
+    accordion.setAttribute('data-show-all-attributes', JSON.stringify(object))
```

Comment:
Could this technique (of setting the required attributes on the fixture in the test itself, rather than the markup at the top of the file) be used in the previous test as well? It makes the test more self-contained and readable.